File: Assets/Scripts/UIPanel/Card/View/CardDetailView.cs
///
/// 抽卡界面的子控件 卡的显示逻辑
///
using UnityEngine;
using UnityEngine.UI;

public class CardDetailView : BasePanel
{
    //UI控件
    private Transform Star;
    private Image Type;
    private Text Name;
    private Image image;

    public CardDetailView(GameObject panel, Transform uiRoot)
    {
        gameObject = panel;
        OpenPanel(uiRoot);
        Init();
    }

    //打开逻辑
    public override void OpenPanel(Transform uiRoot)
    {
        gameObject = GameObject.Instantiate(gameObject, uiRoot, false);
        transform = gameObject.transform;
    }

    public override void ClosePanel()
    {
    }

    //初始化
    private void Init()
    {
        Star = transform.Find("Star");
        Type = transform.Find("Type").GetComponent<Image>();
        Name = transform.Find("Name").GetComponent<Text>();
        image = transform.GetComponent<Image>();
    }
    
    //刷新逻辑
    public void Refresh(HeroLocalItem heroLocalData,Hero hero)
    {
        //显示类别图片
        Texture2D pic = (Texture2D)Resources.Load("Icon/" + hero.type.ToString());
        Sprite tmp = Sprite.Create(pic, new Rect(0, 0, pic.width, pic.height), new Vector2(0, 0));
        Type.sprite = tmp;
        //显示自身图片
        Texture2D pic1 = (Texture2D)Resources.Load(hero.ImgPath);
        Sprite tmp1 = Sprite.Create(pic1, new Rect(0, 0, pic1.width, pic1.height), new Vector2(0, 0));
        image.sprite = tmp1;
        //显示姓名
        Name.text = hero.name;
        //更新星星
        RefreshStars(hero);
    }

    public void RefreshStars(Hero hero)
    {
        for (int i = 0; i < Star.childCount; i++)
        {
            Transform uistar = Star.GetChild(i);
            if ((int)(hero.rarity) > i)
            {
                uistar.gameObject.SetActive(true);
            }
            else
            {
                uistar.gameObject.SetActive(false);
            }
        }
    }
}

File: Assets/LuaScripts/Panel/CardPanel/View/CardDetailView.lua
local CardDetailView = BaseClass("CardDetailView")
-- 初始化函数
function CardDetailView:__init(resource,root)
    --基本属性
    self.controlPanel=GameObject.Instantiate(resource,root,false)
    self.transform=self.controlPanel.transform
    self.transform.localScale=Vector3(0.8,0.8,0.8)

    --是否激活
    self.isActive=false
    --获取组件
    self.star = self.transform:Find("Star");
    self.type = self.transform:Find("Type"):GetComponent("Image");
    self.name = self.transform:Find("Name"):GetComponent("Text");
    self.image = self.transform:GetComponent("Image");
end

--显示逻辑
function CardDetailView:Refresh(heroData)
    --类别图片
    local path=GetName(typeof(HeroType),heroData.type)
    local pic=Resources.Load("Icon/"..tostring(path),typeof(Texture2D))
    self.type.sprite= Sprite.Create(pic, Rect(0, 0, pic.width, pic.height), Vector2(0, 0))
    --英雄图片
    local pic1=Resources.Load(tostring(heroData.imgPath),typeof(Texture2D))
    self.image.sprite= Sprite.Create(pic1, Rect(0, 0, pic1.width, pic1.height), Vector2(0, 0))
    --显示姓名
    self.name.text=heroData.name
    --更新星星
    RefreshStars(self.star,heroData)
end

return CardDetailView